c++/src/kj/units.h

```diff
@@ -347,21 +374,31 @@ class Quantity {
 
   template <typename Number1, typename Number2, typename Unit2>
   friend inline constexpr auto operator*(Number1 a, Quantity<Number2, Unit2> b)
-      -> Quantity<decltype(Number1(1) * Number2(1)), Unit2>;
+      -> Quantity<decltype(Number1() * Number2()), Unit2>;
 
   template <typename T>
   friend inline constexpr T unit();
```

Comment:
Turns out this line is vestigial and can just be deleted. I guess GCC and Clang ignore the friend declaration if it doesn't match any real declaration.